lib/charms/opensearch/v0/opensearch_health.py

```diff
@@ -58,42 +61,87 @@ def apply(
         app: bool = True,
     ) -> str:
         """Fetch cluster health and set it on the app status."""
-        try:
-            host = self._charm.unit_ip if use_localhost else None
-            status = self._fetch_status(host, wait_for_green_first)
-            if not status:
-                return HealthColors.UNKNOWN
-
-            # the health depends on data nodes, for large deployments: an ML cluster
-            # may not be concerned about reporting or relying on the health of the
-            # data nodes in other clusters. We should therefore get this info from
-            # the deployment descriptor which has an overview of all the cluster
-            if not (deployment_desc := self._charm.opensearch_peer_cm.deployment_desc()):
-                return HealthColors.UNKNOWN
-
-            # compute health only in clusters where data nodes exist
-            compute_health = (
-                deployment_desc.start == StartMode.WITH_GENERATED_ROLES
-                or "data" in deployment_desc.config.roles
-            )
-            if not compute_health:
-                return HealthColors.IGNORE
+        status = self.get(
+            wait_for_green_first=wait_for_green_first,
+            use_localhost=use_localhost,
+        )
+        logger.info(f"Current health of cluster: {status}")
+
+        if app:
+            self._apply_for_app(status)
```

Comment:
So, this OpenSearchHealth.apply, if calling apply, there is a potential to call the "trigger defer" method. We must make sure if: (1) we need this "trigger defer" in "_apply_for_app"; or (2) if we do, then we need to ensure there is no risk for any OpenSearchHealth.apply() to be called before a defer

Reply:
This will only happen in the case where, a NON-leader unit wants to apply the health to application status (which it cannot, so it emits this event so that the leader is notified that it must compute it). 

and this happens in exactly 2 cases:
- `_on_opensearch_data_storage_detaching`
- `_post_start_init`